dynamiqs/utils/quantum_utils/general.py

```diff
@@ -975,6 +975,25 @@ def entropy_vn(x: ArrayLike) -> Array:
     return -(w * jnp.log(w)).sum(-1)
 
 
+def eig_callback_cpu(x: Array) -> tuple[Array, Array]:
+    """Wraps jnp.linalg.eig so that it can be jit-ed on a machine with GPUs.
+    See https://github.com/google/jax/issues/1259#issuecomment-2112907161.
+    """
+
+    def _eig_cpu(_x: Array) -> tuple[Array, Array]:
+        # Force this computation to be performed on the cpu by jit-ing and
+        # explicitly specifying the device.
+        with jax.default_device(jax.devices('cpu')[0]):
+            return jax.jit(jnp.linalg.eig)(_x)
+
+    return jax.pure_callback(
+        _eig_cpu,
+        (jnp.ones(x.shape[:-1], dtype=cdtype()), jnp.ones(x.shape, dtype=cdtype())),
+        x.astype(cdtype()),
```

Comment:
Why doesn't the following work?
```suggestion
        x,
```